pkg/minishift/setup/hypervisor/virtualization_darwin.go

```diff
@@ -17,15 +17,80 @@ limitations under the License.
 package hypervisor
 
 import (
+	"errors"
 	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/user"
+
+	"github.com/minishift/minishift/pkg/util/progressdots"
+)
+
+const (
+	driverBinaryDir   = "/usr/local/bin"
+	driverBinaryPath  = driverBinaryDir + "/docker-machine-driver-xhyve"
+	driverDownloadUrl = "https://github.com/machine-drivers/docker-machine-driver-xhyve/releases/download/v0.3.3/docker-machine-driver-xhyve"
 )
 
 func CheckHypervisorAvailable() error {
-	fmt.Println("Checking Hypervisor Availability ...")
 	return nil
 }
 
 func CheckAndConfigureHypervisor() error {
-	fmt.Println("Checking and Configuring Hypervisor  ...")
+	if isRoot() {
+		fmt.Print("Configuring Xhyve Hypervisor.")
+		progressDots := progressdots.New()
+		progressDots.Start()
+		err := downloadXhyveDriver(driverBinaryPath, driverDownloadUrl)
+		progressDots.Stop()
+		fmt.Print("OK")
+		return err
+	}
+	return errors.New("You need to execute this command with sudo.")
+}
+
+func downloadXhyveDriver(filepath string, url string) error {
+	os.MkdirAll(driverBinaryDir, 0751)
+
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return err
+	}
+
+	err = out.Chown(0, 0)
+	if err != nil {
+		return err
+	}
+
+	err = out.Chmod(os.ModeSetuid | 0751)
+	if err != nil {
+		return err
+	}
```

Comment:
We definitely need to add more logging in this function. We should add to the verbose logs `-v2` if something might not be ideal to be shown on the console by default. Also We need to make sure when failure happens, user should be able to pinpoint the failure, so that he can fix it or get better understanding about the failure.

Reply:
WIll add logging to the console, for the user to get an idea of what's being done by the command.